content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesHelperAction.java

```diff
@@ -1880,14 +1872,74 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
 		
 		if(fullPath != null)
 		{
-			String uploadMax = ServerConfigurationService.getString("content.upload.max");
-			String siteQuota = ServerConfigurationService.getString("siteQuota@org.sakaiproject.content.api.ContentHostingService");
 			Long fileSize = Long.parseLong(request.getHeader("content-length"));
-			if((uploadMax != null && !"".equals(uploadMax)) &&   (fileSize /1024L / 1024L) > Long.parseLong(uploadMax)){
+			Long uploadMax;
+			Long siteQuota;
+			
+			try
+			{
+				uploadMax = Long.parseLong( ServerConfigurationService.getString( ResourcesConstants.SAK_PROP_MAX_UPLOAD_FILE_SIZE, 
+																					ResourcesConstants.DEFAULT_MAX_FILE_SIZE_STRING ) );
+			}
+			catch( NumberFormatException ex )
+			{
+				logger.debug( "sakai.property '"+ ResourcesConstants.SAK_PROP_MAX_UPLOAD_FILE_SIZE + "' does not contain an integer", ex );
+				uploadMax = ResourcesConstants.DEFAULT_MAX_FILE_SIZE;
+			}
+			
+			try
+			{
+				siteQuota = Long.parseLong( ServerConfigurationService.getString( ResourcesConstants.SAK_PROP_MASTER_SITE_QUOTA, 
```

Comment:
It would be nice if `ServerConfigurationService` supported long values internally).